File: cpp_solutions/Array/Palindrome_Number.cpp
/*
* Determine whether an integer is a palindrome. Do this without extra space.
* 
* Notes: Mathmatic calculation
**/

bool isPalindrome(int x){
	if(x < 0) return false;
	long rev = 0, temp = 0;
	temp = x;

	while(temp){
		rev = rev*10 + temp % 10;
		temp = temp / 10;
	}

	return rev == x;
}

File: python_solutions/Array/Palindrome_Number.py
class Solution:
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        ret = 0
        temp = x
        while x:
            ret = ret*10 + x%10
            x = int(x/10)
        return temp == ret
        